cmd/dep/ensure.go

```diff
@@ -215,23 +213,23 @@ func (cmd *ensureCommand) Run(ctx *ctx, args []string) error {
 		return errors.Wrap(err, "ensure Solve()")
 	}
 
-	sw := safeWriter{
-		root: p.absroot,
-		m:    p.m,
-		l:    p.l,
-		nl:   solution,
-		sm:   sm,
+	sw := dep.SafeWriter{
+		Root:          p.AbsRoot,
+		Manifest:      p.Manifest,
+		Lock:          p.Lock,
+		NewLock:       solution,
+		SourceManager: sm,
 	}
 
 	// check if vendor exists, because if the locks are the same but
 	// vendor does not exist we should write vendor
 	var writeV bool
-	vendorExists, _ := isDir(filepath.Join(sw.root, "vendor"))
+	vendorExists, _ := dep.IsDir(filepath.Join(p.AbsRoot, "vendor"))
```

Comment:
The diff shows that this should be `sw.Root` instead of `p.AbsRoot`. If this is done for a correctness or bug fix I'd much rather do that as a separate PR after this one.